arbnode/batch_poster.go

```diff
@@ -214,12 +214,12 @@ func NewBatchPoster(l1Reader *headerreader.HeaderReader, inbox *InboxTracker, st
 
 // checkRevert checks blocks with number in range [from, to] whether they
 // contain reverted batch_poster transaction.
-func (b *BatchPoster) checkReverts(ctx context.Context, from, to *big.Int) (bool, error) {
-	if from.Cmp(to) == 1 {
-		return false, fmt.Errorf("wrong range, from: %v is more to: %v", from.Int64(), to.Int64())
+func (b *BatchPoster) checkReverts(ctx context.Context, from, to int64) (bool, error) {
+	if from > to {
+		return false, fmt.Errorf("wrong range, from: %d is more to: %d", from, to)
 	}
-	one := big.NewInt(1)
-	for number := new(big.Int).Set(from); number.Cmp(to) != 1; number.Add(number, one) {
+	for idx := from; idx <= to; idx++ {
+		number := big.NewInt(idx)
 		block, err := b.l1Reader.Client().BlockByNumber(ctx, number)
 		if err != nil {
 			return false, fmt.Errorf("getting block: %v by number: %w", number.Int64(), err)
```

Comment:
FYI big ints have a String method that lets them work nicely with %v, so you can just do `number` instead of `number.Int64()`. Or you could do `idx` and then collapse the `number := big.NewInt(idx)` into the `BlockByNumber` call

Reply:
Done.